packages/dev/core/src/Engines/webgpuEngine.ts

```diff
@@ -666,16 +692,8 @@ export class WebGPUEngine extends AbstractEngine {
         (this.uniqueId as number) = WebGPUEngine._InstanceId++;
         this._glslangOptions = glslangOptions;
         this._twgslOptions = twgslOptions;
-        return this._initGlslang(glslangOptions ?? this._options?.glslangOptions)
-            .then((glslang: any) => {
-                this._glslang = glslang;
-                this._tintWASM = WebGPUEngine.UseTWGSL ? new WebGPUTintWASM() : null;
-                return this._tintWASM
-                    ? this._tintWASM.initTwgsl(twgslOptions ?? this._options?.twgslOptions).then(() => {
-                          return navigator.gpu!.requestAdapter(this._options);
-                      })
-                    : navigator.gpu!.requestAdapter(this._options);
-            })
+        return navigator

```

Comment:
I think we should add an option in the WebGPU engine creation that forces glslang / tintWASM to be loaded at initialization time, to avoid freezes when these packages need to be loaded during program execution. People would probably rather spend a few extra seconds on initialization than suffer freezes at runtime.